src/sds_data_model/dataframe.py

```diff
@@ -0,0 +1,254 @@
+        else:
+            read_file_kwargs = {}
+
+        file_reader_pandas = {
+            ".xlsx": read_excel,
+            ".xls": read_excel,
+            ".xlsm": read_excel,
+            ".xlsb": read_excel,
+            ".odf": read_excel,
+            ".ods": read_excel,
+            ".odt": read_excel,
+        }
+
+        file_reader_spark = {
+            ".csv": _spark.read.csv,
+            ".json": _spark.read.json,
+            ".parquet": _spark.read.parquet,
+        }
+
+        suffix_data_path = Path(data_path).suffix
+
+        if suffix_data_path in file_reader_pandas.keys():
+            data = file_reader_pandas[suffix_data_path](data_path, **read_file_kwargs)
+            data = data.to_spark()
+
+        elif suffix_data_path in file_reader_spark.keys():
+            data = file_reader_spark[suffix_data_path](data_path, **read_file_kwargs)
+        elif suffix_data_path == ".gpkg":
+            data = read_gpkg(data_path, **read_file_kwargs)
+        else:
+            data = read_vector_files(
+                data_path,
+                **read_file_kwargs
+            )
+                  
+        metadata = _get_metadata(
+            data_path=data_path,  
+            metadata_path=metadata_path)
+        
+        _name = _get_name(
+            name=name,
+            metadata=metadata,
+        )
+                
+        return cls(
+            name=_name,
+            data=data,
+            metadata=metadata
+        )
+
+    def call_method(
+        self: Self, 
+        method_name: str,
+        /,
+        *args,
+        **kwargs,
+    ) -> Optional[Self]:
+        """Calls spark method specified by user on spark dataframe in wrapper, using user specified arguments. 
+        THe function:
+        1) assign method call to attribute object, so you can examine it before implementing anything
+        2) check if method_name is a method
+        3) get the signature of method (what argument it takes, return type, etc. )
+        4) bind arguments to function arguments
+        5) create string to pass through graph generator
+        6) return value, as have checked relevant information and can now call method
+        7) if not method, assume its a property (eg. crs)
+        8) check if method_name returns a dataframe, if so update self.data attribute with that new data
+        9) if doesn't return a dataframe, (eg. called display), don't want to overwrite data attribute with nothing, so return original data
+        
+        Examples:
+            >>> from sds_data_model.dataframe import DataFrameWrapper
+            >>> Wrapped = DataFrameWrapper.from_files(name = "National parks",  
+                        data_path="/dbfs/mnt/base/unrestricted/source_defra_data_services_platform/dataset_traditional_orchards/format_SHP_traditional_orchards/LATEST_traditional_orchards/", 
+                        read_file_kwargs = {'suffix':'.shp',  'ideal_chunk_size':1000})
```

Comment:
- Variables should be lowercase, only classes are capitalised in Python.
- You don't need the `ideal_chunk_size`, it's only relevant if you're using the rows concurrency strategy, which you aren't, and it's 1,000 by default anyway.

```suggestion
            >>> wrapped = DataFrameWrapper.from_files(
                name = "National parks",
                data_path="/dbfs/mnt/base/unrestricted/source_defra_data_services_platform/dataset_traditional_orchards/format_SHP_traditional_orchards/LATEST_traditional_orchards/",
                read_file_kwargs = {'suffix':'.shp'})
```